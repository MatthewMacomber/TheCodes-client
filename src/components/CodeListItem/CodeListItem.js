import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import textImage from 'text-to-image';
import {NiceDate, Hyph} from '../Utils/Utils';
import './CodeListItem.css';

export default class CodeListItem extends Component {
  state = {
    codeImage: ''
  };

  setCodeImage = codeImage => {
    this.setState({codeImage})
  };

  componentDidMount() {
    const {code} = this.props;
    textImage.generate(code.content, {
      maxWidth: 300,
      customHeight: 180,
      fontSize: 50,
      lineHeight: 55,
      textAlign: 'start',
      bgColor: 'lightgrey',
      textColor: 'black'
    })
      .then(this.setCodeImage);
  };

  render() {
    const {code} = this.props;
    let codeImage = this.state.codeImage;
    return (
      <Link to={`/code/${code.id}`} className='CodeListItem'>
        <img className="image" src={codeImage} alt={code.content}/>
        <div className="overlay after">
          <header className='CodeListItem__header'>
            <h2 className='CodeListItem__heading'>
              {code.title}
            </h2>
            <CodeDate code={code} />
          </header>
          <footer className='CodeListItem__footer'>
            {code.user_name && <>
              Created by
              <Hyph />
              <CodeAuthor code={code} />
            </>}
          </footer>
        </div>
      </Link>
    );
  };
};

function CodeDate({code}) {
  return (
    <span className='CodeListItem__date'>
      <NiceDate
        date={code.date_created}
      />
    </span>
  );
};

function CodeAuthor({code}) {
  return (
    <span className='CodeListItem__author'>
      {code.user_name}
    </span>
  );
};
