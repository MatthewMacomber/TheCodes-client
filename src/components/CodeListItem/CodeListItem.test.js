import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import CodeListItem from './CodeListItem';

describe('CodeListItem component', () => {
  //const props = {}
});