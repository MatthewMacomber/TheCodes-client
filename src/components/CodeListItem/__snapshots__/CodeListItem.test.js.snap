// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeListItem component renders the CodeListItem given props 1`] = `
<Link
  className="CodeListItem"
  to="/code/1"
>
  <img
    alt="test code"
    className="image"
    src=""
  />
  <div
    className="overlay after"
  >
    <header
      className="CodeListItem__header"
    >
      <h2
        className="CodeListItem__heading"
      >
        code
      </h2>
      <CodeDate
        code={
          Object {
            "content": "test code",
            "date_created": 2019-01-15T08:00:00.000Z,
            "id": 1,
            "title": "code",
            "user_name": "test",
          }
        }
      />
    </header>
    <footer
      className="CodeListItem__footer"
    >
      Created by
      <Hyph />
      <CodeAuthor
        code={
          Object {
            "content": "test code",
            "date_created": 2019-01-15T08:00:00.000Z,
            "id": 1,
            "title": "code",
            "user_name": "test",
          }
        }
      />
    </footer>
  </div>
</Link>
`;
